cli: TestZipIncludeAndExcludeFilesDataDriven failed

cli.TestZipIncludeAndExcludeFilesDataDriven [failed](https://mesolite.cluster.engflow.com/invocations/default/6ed08ce0-51b9-4031-9a17-e59181dee830?testReportRun=2&testReportShard=8&testReportAttempt=1#targets-Ly9wa2cvY2xpOmNsaV90ZXN0) on release-25.2.12-rc @ [7fc1e95228ad6610c2ea21d8ba55d0d6513c319f](https://github.com/cockroachdb/cockroach/commits/7fc1e95228ad6610c2ea21d8ba55d0d6513c319f):


```
debug/system.namespace.txt
debug/system.privileges.txt
debug/system.protected_ts_meta.txt
debug/system.protected_ts_records.txt
debug/system.rangelog.txt
debug/system.replication_constraint_stats.txt
debug/system.replication_critical_localities.txt
debug/system.replication_stats.txt
debug/system.reports_meta.txt
debug/system.role_id_seq.txt
debug/system.role_members.txt
debug/system.role_options.txt
debug/system.scheduled_jobs.txt
debug/system.settings.txt
debug/system.span_configurations.txt
debug/system.sql_instances.txt
debug/system.sql_stats_cardinality.txt
debug/system.sqlliveness.txt
debug/system.statement_diagnostics.txt
debug/system.statement_diagnostics_requests.txt
debug/system.statement_statistics_limit_5000.txt
debug/system.table_statistics.txt
debug/system.task_payloads.txt
debug/system.tenant_settings.txt
debug/system.tenant_tasks.txt
debug/system.tenant_usage.txt
debug/system.tenants.txt
debug/system.zones.txt
debug/tenant_ranges.err.txt
    zip_test.go:1505: -- test log scope end --
    zip_test.go:1443: 
          testdata/zip/file-filters/testzip_file_filters:439:
          --exclude-files=*.json
        output didn't match expected:
          @@ -32,11 +32,13 @@
           debug/crdb_internal.transaction_contention_events.txt
           debug/crdb_internal.zones.txt
           debug/debug_zip_command_flags.txt
           debug/hot-ranges-tenant.sh
           debug/hot-ranges.sh
          -debug/nodes/1/cpu.pprof
          +debug/nodes/1/cpu.pprof.err.txt
          +debug/nodes/1/cpuprof/cpuprof.2026-01-27T08_53_56.910.67.pprof
          +debug/nodes/1/cpuprof/cpuprof.2026-01-27T08_54_07.416.89.pprof
           debug/nodes/1/crdb_internal.active_range_feeds.txt
           debug/nodes/1/crdb_internal.cluster_replication_node_stream_checkpoints.txt
           debug/nodes/1/crdb_internal.cluster_replication_node_stream_spans.txt
           debug/nodes/1/crdb_internal.cluster_replication_node_streams.txt
           debug/nodes/1/crdb_internal.feature_usage.txt
--- FAIL: TestZipIncludeAndExcludeFilesDataDriven/testzip_file_filters (162.65s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>deadlock=true</code>
 - <code>run=2</code>
 - <code>shard=8</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #161267 cli: TestZipIncludeAndExcludeFilesDataDriven failed [C-test-failure O-robot T-observability branch-release-26.1 s390x-test-failure]
</p>
</details>
/cc @cockroachdb/obs-prs @cockroachdb/server @cockroachdb/obs-india-prs
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestZipIncludeAndExcludeFilesDataDriven.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-59115

--- a/pkg/cli/zip_test.go
+++ b/pkg/cli/zip_test.go
@@ -1496,6 +1496,12 @@ func trimNonDeterministicZipOutputFiles(out string) string {
 	out = re.ReplaceAllString(out, ``)
 	re = regexp.MustCompile(`(?m).*goroutine_dump.*\.pb\.gz$` + "\n")
 	out = re.ReplaceAllString(out, ``)
+	// CPU profiles are non-deterministic: stored profiles have timestamps in
+	// filenames, and live profile collection may fail producing error files.
+	re = regexp.MustCompile(`(?m).*cpuprof/cpuprof\..*\.pprof$` + "\n")
+	out = re.ReplaceAllString(out, ``)
+	re = regexp.MustCompile(`(?m).*cpu\.pprof\.err\.txt$` + "\n")
+	out = re.ReplaceAllString(out, ``)
 	return out
 }
 

